lib/frozen_record/base.rb

```diff
@@ -195,6 +204,9 @@ def load_records(force: false)
 
         @records ||= begin
           records = backend.load(file_path)
+          if attribute_deserializers
+            records = records.map { |r| deserialize_attributes!(r.dup) }
```

Comment:
It annoys me a bit that we dup the records several times.

We may need to refactor this to avoid the multiple dup, or use a `dup if frozen` or something.

Also you are not re-freezing it here.

Reply:
I refactored it so it's at least no worse than it was before (I think).  

I tried to refactor further, getting rid of another `.map` and `.dup` by removing 
```
if attribute_deserializers.any? || default_attributes
  records = records.map { |r| assign_defaults!(deserialize_attributes!(r.dup)).freeze }.freeze
end
```

and changing 
```diff
-  records = FrozenRecord.ignore_max_records_scan { records.map { |r| load(r) }.freeze }
+  records = FrozenRecord.ignore_max_records_scan { records.map { |r| new(r.dup).freeze }.freeze }
```
but that required changes to listing attributes and the change got too big.   I think it could be done but maybe in a separate PR?